scripts/translate_challenge.py

```
@@ -0,0 +1,261 @@
+        }
+
+    # Parse XML template to create BlockSegment
+    try:
+        # Use decode_friendly to get the template element
+        template_element = None
+        for elem in decode_friendly(xml_template, tags="xml"):
+            template_element = elem
+            break
+
+        if template_element is None:
+            raise ValueError("No XML element found in template")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)